lms/djangoapps/discussion/rest_api/api.py

```diff
@@ -383,7 +384,23 @@ def get_courseware_topics(
     for xblock in discussion_xblocks:
         xblocks_by_category[xblock.discussion_category].append(xblock)
 
-    for category in xblocks_by_category.keys():
+    def sort_categories(category_list):
+        """
+        Sorts the given iterable containing alphanumeric correctly.
+        Required arguments:
+        category_list -- list of categories.
+        """
+        def convert(text):
+            if text.isdigit():
+                return int(text)
+            return text
+
+        def alphanum_key(key):
+            return [convert(c) for c in re.split('([0-9]+)', key)]
+
+        return sorted(category_list, key=alphanum_key)
+
+    for category in sort_categories(xblocks_by_category.keys()):
```

Comment:
This will sort the topics in alphabetical order which is not required. The expectation is that the topics should have the same order as they have in the course which will not always be alphabetical.